<?xml version="1.0"?>

<launch>

    <arg name="des_x" default= "-4"/>
    <arg name="des_y" default= "7"/>
    <arg name="state" default= "0"/>
    <arg name="time" default= "0"/>
    <arg name='bug_0' default= "0"/>
    <param name="des_pos_x" value="$(arg des_x)" />
    <param name="des_pos_y" value="$(arg des_y)" />
    <param name="state_value" value="$(arg state)" />
    <param name="target_time" value="$(arg time)"/>
    <param name="bug_trigger" value="$(arg bug_0)"/>
    
    <!-- create the nodes to the bug zero algorithm -->
    <node pkg="final_assignment" type="wall_follow_service_m.py" name="wall_follower" />
    <node pkg="final_assignment" type="go_to_point_service_m.py" name="go_to_point" /> 

    <!-- create the node for the move_base algorithm including the launch file-->
    <include file="$(find final_assignment)/launch/move_base.launch"></include>

    <!-- create the user interface -->
    <node pkg="final_assignment" type="user_interface.py" name="user_interface" output="screen" />

    <!-- run the server node for providing the input in case of random choice  -->
    <node pkg="final_assignment" type="position_server.py" name="target_provider" output="screen" />

    <!-- run the main node  -->
    <node pkg="final_assignment" type="main_m.py" name="main" output="screen" />

</launch>
